src/main/scala/com/actian/spark_vector/datastream/reader/DataStreamTap.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 Actian Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.actian.spark_vector.datastream.reader
+
+import org.apache.spark.Logging
+
+import com.actian.spark_vector.util.ResourceUtil.closeResourceOnFailure
+import com.actian.spark_vector.colbuffer.ColumnBuffer
+
+import java.nio.ByteBuffer
+import java.nio.channels.SocketChannel
+
+/** The `VectorTap` that reads a `Vector DataStream` from a socket as a `ByteBuffer` */
+private[reader] case class DataStreamTap(implicit val socket: SocketChannel) extends Logging with Serializable {
+  import DataStreamReader._
+
+  private final val BinaryDataCode = 5 /* X100CPT_BINARY_DATA_V2 */
+  private final val NumTuplesIndex = 4
+
+  private var vectors: ByteBuffer = _
+  private var tapOpened: Boolean = true
```

Comment:
can we call this isNextVectorBuffered = false and set it to true when isEmpty is called?